static/js/tab_bar.js

```diff
@@ -168,6 +168,14 @@ exports.render_title_area = function () {
 exports.maybe_rerender_title_area_for_stream = function (old_sub) {
     // TODO: Implement rerendering for subscriber count changes.
     // We simply need to call this function in the appropriate places.
+
+    // This should never happen, but it's better to validate the assumption
+    // that old_sub exists and throw an error and then exit if not
+    if (typeof old_sub === "undefined") {
+        blueslip.error('Undefined sub passed to function tab_bar.maybe_rerender_title_area_for_stream');
+        return;
+    }
```

Comment:
I'm not sure this adds something over just throwing an exception 2 lines down.  I suppose it doesn't do much harm, but I'm curious why you decided to add this logic?

In any case, I think the local variable should be called `sub`, not `old_sub`; or maybe `modified_sub` if you want to be explicit about its role here.

Reply:
(In our future typescript world I'd expect this to be verified by the time system and us to delete this check)

Reply:
I added it because I thought throwing a blueslip error would be better than a JS exception if this ends up being called incorrectly in the future...
... but I wasn't sure if it's definitely better so I put it on a separate commit that we can just drop.

Reply:
(btw, I changed the variable to `modified_sub`)